### PR TITLE
Migrate to terraform-plugin-framework

### PR DESCRIPTION
**Pull request by bendbennett**
Closes: #112

### CODE REVIEW

File: internal/provider/resource_archive_file.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+	model.ID = types.StringValue(sha1)
+
+	return diags
+}
+
+func (d *archiveFileResource) Update(_ context.Context, _ resource.UpdateRequest, _ *resource.UpdateResponse) {
+}
+
+func (d *archiveFileResource) Delete(_ context.Context, _ resource.DeleteRequest, _ *resource.DeleteResponse) {
+}
```
Comment (bflad) [inline internal/provider/resource_archive_file.go:268]: I really wish we could "finish" this resource's implementation and un-deprecate it with deleting the archive file, however that could only be considered in a breaking change for a major version. 🙁

File: internal/provider/resource_archive_file.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+							Description: "Add this content to the archive with `filename` as the filename.",
+							Required:    true,
+							Validators: []validator.String{
+								stringvalidator.ConflictsWith(
+									fwpath.MatchRoot("source_file"),
+									fwpath.MatchRoot("source_dir"),
+									fwpath.MatchRoot("source_content"),
+									fwpath.MatchRoot("source_content_filename"),
+								),
+							},
```
Comment (bflad) [inline internal/provider/resource_archive_file.go:49]: Similar note about moving the validators to the set.

**Review by bflad — APPROVED** [review]
Overall this looks really good, just some minor bits to consider.

**Comment by bendbennett** [comment]
Following the moving of the validators onto the set itself and adding the `ConfigValidators` onto the datasource and the resource, the tests fail with errors indicating that `"source"` is specified, when it is not defined in the Terraform configuration. 

For example:
```shell
=== CONT  TestAccArchiveFile_Basic
    data_source_archive_file_test.go:20: Step 1/5 error: Error running pre-apply refresh: exit status 1
        
        Error: Invalid Attribute Combination
        
          with data.archive_file.foo,
          on terraform_plugin_test.tf line 2, in data "archive_file" "foo":
           2: data "archive_file" "foo" {
        
        Attribute "source_content" cannot be specified when "source" is specified
```

The configuration that is used by this test is:
```terraform
data "archive_file" "foo" {
  type                    = "zip"
  source_content          = "This is some content"
  source_content_filename = "content.txt"
  output_path             = "%s"
}
```

**Review by Dezzj — COMMENTED** [review]
Veiwed

**Pull request merged**